### PR TITLE
fix: support array of tableHints and allow joins to use tableHints

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR allows for an array of tableHints to be passed to a query and also allows for any joins to also have the tableHints applied.

As part of this, I have moved the logic for adding index & table hints to `quoteTable` as they need to be appended after table name in the query.

### CODE REVIEW

File: packages/core/test/unit/query-generator/select-query.test.ts [hunk 1]
```diff
@@ -675,4 +676,102 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
...
+
+    it('should not add an index hint if the type is not valid', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        // @ts-expect-error -- we are testing invalid values
+        indexHints: [{ type: 'INVALID', values: ['index_project_on_name'] }],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['indexHints']),
+        'mariadb mysql snowflake': 'SELECT [id] FROM [Users] AS [User];',
```
Comment (WikiRik) [inline packages/core/test/unit/query-generator/select-query.test.ts:711]: Maybe we should throw an error on invalid indexHints?

File: packages/core/test/unit/query-generator/select-query.test.ts [hunk 2]
```diff
@@ -675,4 +676,102 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
...
+
+    it('should not add a table hint if the type is not valid', () => {
+      expectsql(() => queryGenerator.selectQuery(User.table, {
+        model: User,
+        attributes: ['id'],
+        // @ts-expect-error -- we are testing invalid values
+        tableHints: ['INVALID'],
+      }, User), {
+        default: buildInvalidOptionReceivedError('quoteTable', sequelize.dialect.name, ['tableHints']),
+        mssql: 'SELECT [id] FROM [Users] AS [User];',
```
Comment (WikiRik) [inline packages/core/test/unit/query-generator/select-query.test.ts:773]: Same here, maybe we should throw on invalid tableHints?

**Review by WikiRik — APPROVED** [review]
Two small comments but non blocking

File: packages/core/test/unit/query-generator/select-query.test.ts (hunk 1 above)
Comment (lohart13) [reply packages/core/test/unit/query-generator/select-query.test.ts]: Good idea, updated now

File: packages/core/test/unit/query-generator/select-query.test.ts (hunk 2 above)
Comment (lohart13) [reply packages/core/test/unit/query-generator/select-query.test.ts]: Good idea, updated now

**Review by WikiRik — APPROVED** [review]

**Pull request merged**